src/xproc3/harness-lib.xpl

```diff
@@ -0,0 +1,293 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- ===================================================================== -->
+<!--  File:       harness-lib.xpl                                          -->
+<!--  Author:     Florent Georges                                          -->
+<!--  Date:       2011-11-08                                               -->
+<!--  Contributors:                                                        -->
+<!--        George Bina - updated to use XProc 3                           -->
+<!--  Date:       2025-06-09                                               -->
+<!--  URI:        http://github.com/xspec/xspec                            -->
+<!--  Tags:                                                                -->
+<!--    Copyright (c) 2011 Florent Georges (see end of file.)              -->
+<!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->
+
+
+<p:library xmlns:c="http://www.w3.org/ns/xproc-step"
+           xmlns:p="http://www.w3.org/ns/xproc"
+           xmlns:pkg="http://expath.org/ns/pkg"
+           xmlns:x="http://www.jenitennison.com/xslt/xspec"
+           xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
+           xmlns:xs="http://www.w3.org/2001/XMLSchema"
+           xmlns:map="http://www.w3.org/2005/xpath-functions/map"
+           pkg:import-uri="#none"
+           exclude-inline-prefixes="map xs xsl x pkg p c"
+           version="3.1">
+
+   
+   <!--
+       Pass through and possibly log the input.
+       
+       If there is a parameter with the name $if-set, its value must be a URI, where
+       to log the input.  If there is not, then no log is produced.
+   -->
+   <p:declare-step type="x:log" name="log">
+      <!-- the port declarations -->
+      <p:input  port="source" primary="true"/>
+      <p:output port="result" primary="true"/>
+      
+      <p:option name="parameters" as="map(xs:QName,item()*)?"/>
+      <p:option name="if-set" required="true"/>
+      
+      <p:group>
+         <p:variable name="uri" select="map:get($parameters, xs:QName($if-set))"/>
+         <p:choose>
+            <p:when test="$uri">
+               <p:store serialization="map{'method':'adaptive'}">
+                  <p:with-option name="href" select="$uri"/>
```

Comment:
What do you think about having `p:store` on line 45 say what it is doing, e.g., `message="[x:log] Saving {$if-set} to {$uri}."`? Is that useful or will it just be clutter in the console?